addons/mail/static/src/messaging/component/notification_list/notification_list.js

```diff
@@ -38,9 +39,9 @@ class NotificationList extends Component {
     get notifications() {
         const { notifications } = this.storeProps;
         return notifications.map(notification =>
-            notification.type === 'thread'
+            notification.thread
                 ? Object.assign({}, notification, {
-                    thread: this.env.entities.Thread.get(notification.uniqueId),
+                    thread: this.env.entities.Thread.get(notification.thread),
```

Comment:
done (removed this get on thread, I think it's safe because there is no async between a useStore and following call to this getter. If we are not confident enough, could reintroduce this ternary condition?)